mllib/src/main/scala/org/apache/spark/ml/clustering/KMeans.scala

```diff
@@ -320,6 +368,23 @@ class KMeans @Since("1.5.0") (
       .setMaxIterations($(maxIter))
       .setSeed($(seed))
       .setEpsilon($(tol))
+
+    if (isDefined(initialModel)) {
+      // Check that the feature dimensions are equal
+      val dimOfData = rdd.first().size
+      val dimOfInitialModel = $(initialModel).clusterCenters.head.size
+      require(dimOfData == dimOfInitialModel,
+        s"mismatched dimension, $dimOfData in data while $dimOfInitialModel in the initial model.")
+
+      // Check that the number of clusters are equal
+      val kOfInitialModel = $(initialModel).parentModel.clusterCenters.length
+      require(kOfInitialModel == $(k),
```

Comment:
I'd recommend that this log a warning instead of causing a failure.  If we use CrossValidator to select amongst initial models, then